Beistellung_Implementierung/system/Coding/src/app/StateHandler.c

```diff
@@ -0,0 +1,267 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2024   -   www.newtec.de
+***************************************************************************************************/
+/**
+ * @file       StateHandler.c
+ *
+ *    Decides which states are called next depending on the returned events  
+ */
+/**************************************************************************************************/
+
+/* INCLUDES ***************************************************************************************/
+#include "StateHandler.h"
+#include "EventEnum.h"
+
+#include "app/State/InitializeMcu.h"
+#include "app/State/CalibrateLineSensors.h"
+#include "app/State/ReadyToDrive.h"
+#include "app/State/DriveToStart.h"
+#include "app/State/DriveToFinish.h"
+#include "app/State/DriveOverGap.h"
+#include "app/State/DisplayCountdown.h"
+#include "app/State/DisplayLapTime.h"
+#include "app/State/ErrorHandlerState.h"
+#include "app/State/SetParameters.h"
+/* CONSTANTS **************************************************************************************/
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+/** The next state is decided on the basis of the transferred parameter 
+ * @param[in] gCurrentEvent Event that occurred in Drive to Start
+*/
+static void driveToStartEvent(EventEnum gCurrentEvent);
```

Comment:
parameter name should not start with g. This is reserved only for static variables.

as a note, you could either directly use the gCurrentEvent static variable or change the parameter name to something else (currently name is identical to static variable).

Reply:
(We already made a lot of changes to the code of the implementation phase for a running final version. Therefore changes won't be adresses directly in this old version but in the new one.)

Now corrected in the new version.